packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -384,103 +387,115 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
       },
     },
 
-    renderChunk(code, _, { format }) {
-      // make sure we only perform the preload logic in modern builds.
-      if (code.indexOf(isModernFlag) > -1) {
-        const re = new RegExp(isModernFlag, 'g')
-        const isModern = String(format === 'es')
-        if (this.environment.config.build.sourcemap) {
-          const s = new MagicString(code)
-          let match: RegExpExecArray | null
-          while ((match = re.exec(code))) {
-            s.update(match.index, match.index + isModernFlag.length, isModern)
-          }
-          return {
-            code: s.toString(),
-            map: s.generateMap({ hires: 'boundary' }),
-          }
-        } else {
-          return code.replace(re, isModern)
+    renderChunk: {
+      handler(code, _, { format }) {
```

Comment:
On the other hand, this `renderChunk` and `generateBundle` doesn't have to be in the `handler`.